### PR TITLE
Modifications to tuningpack code

### PR DESCRIPTION
**Pull request by johanbogg**
As per discussions on trionictuning.com

### CODE REVIEW

File: T8Suite/Form1.cs [hunk 1]
```diff
@@ -8384,6 +8384,45 @@ public SearchReplaceTuningPackage(string searchReplace)
...
+                                    //break the address up in bytes, add each byte to newsearchString.
+                                    string tempBytes = BitConverter.ToString(bSymSearch);
+                                    string[] tempindBytes = tempBytes.Substring(0,tempBytes.Length).Split('-');
+                                    foreach (string bajt in tempindBytes)
+                                    {
+                                        newsearchString.Add("0x" + bajt);
+                                        
+                                    }
+                                    break;
+
```
Comment (mattiasclaesson) [inline T8Suite/Form1.cs]: Remove the extra newlines

File: T8Suite/Form1.cs [hunk 2]
```diff
@@ -8384,6 +8384,45 @@ public SearchReplaceTuningPackage(string searchReplace)
...
+                                    byte[] bSymSearch = { };
+                                    bSymSearch = BitConverter.GetBytes((int)cfsh.Flash_start_address);
+                                    Array.Reverse(bSymSearch);
+                                    //break the address up in bytes, add each byte to newsearchString.
+                                    string tempBytes = BitConverter.ToString(bSymSearch);
+                                    string[] tempindBytes = tempBytes.Substring(0,tempBytes.Length).Split('-');
+                                    foreach (string bajt in tempindBytes)
+                                    {
+                                        newsearchString.Add("0x" + bajt);
+
```
Comment (mattiasclaesson) [inline T8Suite/Form1.cs]: Remove extra new lines

File: T8Suite/Form1.cs [hunk 3]
```diff
@@ -8384,6 +8384,45 @@ public SearchReplaceTuningPackage(string searchReplace)
...
+                            }
+                        }
+                        else
+                        {
+                            //if not a a symbol, add it verbatim to the new string
+                            newsearchString.Add(searchpart);
+                        }
+                    }
+                    //converts newreplacestring to an array and replaces replaceString with it
+                        searchString = newsearchString.ToArray();
```
Comment (mattiasclaesson) [inline T8Suite/Form1.cs]: Fix white spaces

File: T8Suite/Form1.cs [hunk 4]
```diff
@@ -8384,6 +8384,45 @@ public SearchReplaceTuningPackage(string searchReplace)
...
+                 //MY CHANGES STOP HERE
```
Comment (mattiasclaesson) [inline T8Suite/Form1.cs]: Remove this line

File: T8Suite/Form1.cs [hunk 5]
```diff
@@ -8384,6 +8384,45 @@ public SearchReplaceTuningPackage(string searchReplace)
                 else
                 {
                     string[] searchString = inputStr.Substring(foundS1, foundS2 - foundS1).Split(',');
+                    //creates a temporary collection list for the seacrch strings.
```
Comment (mattiasclaesson) [inline T8Suite/Form1.cs]: search

File: T8Suite/Form1.cs [hunk 6]
```diff
@@ -8396,6 +8435,44 @@ public SearchReplaceTuningPackage(string searchReplace)
...
+                                string tempBytes = BitConverter.ToString(bSymReplace);
+                                string[] tempindBytes = tempBytes.Substring(0, tempBytes.Length).Split('-');
+                                foreach (string bajt in tempindBytes)
+                                {
+                                    newreplaceString.Add("0x" + bajt);
+
+                                }
+                                break;
+
+
```
Comment (mattiasclaesson) [inline T8Suite/Form1.cs]: same here with new lines

File: T8Suite/Form1.cs [hunk 7]
```diff
@@ -8396,6 +8435,44 @@ public SearchReplaceTuningPackage(string searchReplace)
                 foundS1 = inputStr.IndexOf('{') + 1;
                 foundS2 = inputStr.IndexOf('}');
                 string [] replaceString = inputStr.Substring(foundS1, foundS2 - foundS1).Split(',');
```
Comment (mattiasclaesson) [inline T8Suite/Form1.cs]: Search and Replace code looks rather similar. Maybe its possible to extract a method that both use?

Comment (johanbogg) [reply T8Suite/Form1.cs]: this is part of the search and replace code, not sure what you  mean :)

Comment (johanbogg) [reply T8Suite/Form1.cs]: Done

**Pull request merged**